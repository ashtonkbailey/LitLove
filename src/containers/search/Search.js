import React, { Component } from 'react';

class Search extends Component {
  constructor() {
    super()
    this.state = {
      title: ''
    }
  }

  handleChange = (e) => {
    // may just sub in 'title' if there aren't any other state items
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  handleSubmit = () => {

  }

  render() {
    const { title } = this.state;

    return (
      <div>
        <p className="search-text">
          Welcome!
          <br/>
          We’d like to help you find your next great read.
          Enter a book title you love, and click the search
          button to find that book. Based on your choice,
          we’ll give you some recommendations.
        </p>
        <form
          className="search-form"
          onSubmit={this.handleSubmit}
        >
          <input
            onChange={this.handleChange}
            type="text"
            name="title"
            value={title}
            placeholder="Book Title"
            autofocus={true}
          />
          <button className="search-btn">
            Start your Search
          </button>
        </form>
      </div>
    )
  }
}

export default Search;
