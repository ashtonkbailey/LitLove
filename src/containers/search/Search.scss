.text {
  font-size: 34px;
  text-align: center;
  color: #FFFFFF;
  width: 700px;
  margin: auto;
}

.welcome {
  height: 300px;
  margin-top: 1em;
}

.search {
  height: 150px;
  margin-top: 3em;
}

.search-form {
  margin: auto;
  width: 300px;
  display: flex;
  flex-direction: column;
  height: 9em;
  justify-content: space-between;
  align-items: center;

  input {
    color: #FFFFFF;
    background: #7B7B7B;
    border: 1px solid #4D4C4C;
    height: 2.5em;
    font-size: 18px;
    width: 15em;
    padding-left: 0.5em;

    &::-webkit-input-placeholder {
      color: #FFFFFF;
      font-size: 18px;
    }

    &::-moz-placeholder {
      color: #FFFFFF;
      font-size: 18px;
    }

  }

  button {
    background: #DAB192;
    box-shadow: 
      0px 4px 4px rgba(0, 0, 0, 0.25),
      inset -5px 6px 15px #EBCBB3;
    font-family: Sarabun;
    font-weight: bold;
    font-size: 18px;
    text-align: center;
    color: #363636;
    height: 2.5em;
    width: 10em;
    outline: none;
    cursor: pointer;
  }

}