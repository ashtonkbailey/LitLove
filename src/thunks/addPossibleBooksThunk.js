import { googleKey } from '../utils/googleAPI.js';
import { addPossibleBooks, setError } from '../actions';

export const addPossibleBooksThunk = (title) => {
  return async (dispatch) => {
    try {
      const url = `https://www.googleapis.com/books/v1/volumes?q=${title}&key=${googleKey}&maxResults=5&fields=items(id, volumeInfo/title, volumeInfo/authors, volumeInfo/description, volumeInfo/imageLinks)`;
      const response = await fetch(url);
      if (!response.ok) {
        throw Error('Unable to get book list')
      }
      const possibleBooks = await response.json();
      dispatch(addPossibleBooks(possibleBooks.items))
    } catch (error) {
      dispatch(setError(error.message))
    }
  }
}
