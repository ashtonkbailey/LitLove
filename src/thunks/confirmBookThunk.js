import { tDKey } from '../utils/tasteDiveAPI.js';
import { addLovedBook, setError } from '../actions';

export const confirmBookThunk = (bookTitle) => {
  return async (dispatch) => {
    try {
      const url = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?q=${bookTitle}&type=books&info=1&k=${tDKey}`;
      const response = await fetch(url);
      if (!response.ok) {
        throw Error('Unable to get book recommendations')
      }
      const recommendedBooks = await response.json();
      console.log(recommendedBooks)
      dispatch(addLovedBook(bookTitle))
    } catch (error) {
      dispatch(setError(error.message))
    }
  }
}
