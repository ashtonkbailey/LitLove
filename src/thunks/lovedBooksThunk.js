import { googleKey } from '../utils/googleAPI.js';
import { addPossibleBooks } from '../actions';

export const lovedBookThunk = async (title) => {
  return async (dispatch) => {
    try {
      const url = `https://www.googleapis.com/books/v1/volumes?q=${title}&key=${googleKey}&fields=items(volumeInfo/title, volumeInfo/authors, volumeInfo/description, volumeInfo/imageLinks)`;
      const response = await fetch(url);
      if (!response.ok) {
        throw Error('Unable to get book list')
      }
      const possibleBooks = await response.json();
      dispatch(addPossibleBooks(possibleBooks))
    }
  }
}