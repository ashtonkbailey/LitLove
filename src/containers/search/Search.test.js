import React from 'react';
import { shallow } from 'enzyme';

import { Search, mapDispatchToProps } from './Search';
import { addPossibleBooksThunk } from '../../thunks/addPossibleBooksThunk';

jest.mock('../../thunks/addPossibleBooksThunk');

describe('Search', () => {
  let wrapper;
  let type;

  it('should match the snapshot with type of "welcome"', () => {
    type = "welcome";
    wrapper = shallow(<Search type={type} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with type of "new"', () => {
    type = "new";
    wrapper = shallow(<Search type={type} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should update state', () => {
      const mockEvent = {target: {name: 'title', value: 'she'}};
      wrapper = shallow(<Search type="new" />);
      const expected = 'she';
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().title).toEqual(expected);
    });
  });

  describe('handleSubmit', () => {
    let mockEvent;

    beforeEach(() => {
      mockEvent = { preventDefault: jest.fn() };
    });

    it('should run handleSubmit on form submit', () => {
      wrapper = shallow(<Search type="new" />);
      const e = { preventDefault: jest.fn() }
      wrapper.instance().handleSubmit = jest.fn();
      wrapper.instance().forceUpdate(); 
      wrapper.find('form').simulate('submit', e);
      expect(wrapper.instance().handleSubmit).toBeCalledWith(e);
    });

    it('should update state', async () => {
      const mockAddPossibleBooks = jest.fn()
      wrapper = shallow(<Search type="new" addPossibleBooks={mockAddPossibleBooks} />);
      const expected = true;
      await wrapper.instance().handleSubmit(mockEvent);
      expect(wrapper.state().submitted).toEqual(expected);
    })
  });

  describe('mapDispatchToProps', () => {
    it('should return an object with the addPossibleBooks method', () => {
      const mockDispatch = jest.fn();
      const title = 'She';
      const thunk = addPossibleBooksThunk(title);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addPossibleBooks(title);
      expect(mockDispatch).toHaveBeenCalledWith(thunk);
    })
  })
    
})
