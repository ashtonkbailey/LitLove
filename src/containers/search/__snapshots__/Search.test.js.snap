// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot with type of "new" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    type="new"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="search text"
        >
          Looking for your next book? Enter a title you love to start the search process.
        </p>,
        <form
          className="search-form"
          onSubmit={[Function]}
        >
          <input
            autoFocus={true}
            name="title"
            onChange={[Function]}
            placeholder="Book Title"
            type="text"
            value=""
          />
          <button
            className="search-btn"
          >
            New Search
          </button>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Looking for your next book? Enter a title you love to start the search process.",
          "className": "search text",
        },
        "ref": null,
        "rendered": "Looking for your next book? Enter a title you love to start the search process.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              autoFocus={true}
              name="title"
              onChange={[Function]}
              placeholder="Book Title"
              type="text"
              value=""
            />,
            <button
              className="search-btn"
            >
              New Search
            </button>,
          ],
          "className": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "name": "title",
              "onChange": [Function],
              "placeholder": "Book Title",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New Search",
              "className": "search-btn",
            },
            "ref": null,
            "rendered": "New Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="search text"
          >
            Looking for your next book? Enter a title you love to start the search process.
          </p>,
          <form
            className="search-form"
            onSubmit={[Function]}
          >
            <input
              autoFocus={true}
              name="title"
              onChange={[Function]}
              placeholder="Book Title"
              type="text"
              value=""
            />
            <button
              className="search-btn"
            >
              New Search
            </button>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Looking for your next book? Enter a title you love to start the search process.",
            "className": "search text",
          },
          "ref": null,
          "rendered": "Looking for your next book? Enter a title you love to start the search process.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                autoFocus={true}
                name="title"
                onChange={[Function]}
                placeholder="Book Title"
                type="text"
                value=""
              />,
              <button
                className="search-btn"
              >
                New Search
              </button>,
            ],
            "className": "search-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "name": "title",
                "onChange": [Function],
                "placeholder": "Book Title",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New Search",
                "className": "search-btn",
              },
              "ref": null,
              "rendered": "New Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Search should match the snapshot with type of "welcome" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    type="welcome"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="welcome text"
        >
          Welcome!
          <br />
          We’d like to help you find your next great read. Enter a book title you love, and click the search button to find that book. Based on your choice, we’ll give you some recommendations.
        </p>,
        <form
          className="search-form"
          onSubmit={[Function]}
        >
          <input
            autoFocus={true}
            name="title"
            onChange={[Function]}
            placeholder="Book Title"
            type="text"
            value=""
          />
          <button
            className="search-btn"
          >
            Start your Search
          </button>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome!",
            <br />,
            "We’d like to help you find your next great read. Enter a book title you love, and click the search button to find that book. Based on your choice, we’ll give you some recommendations.",
          ],
          "className": "welcome text",
        },
        "ref": null,
        "rendered": Array [
          "Welcome!",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "We’d like to help you find your next great read. Enter a book title you love, and click the search button to find that book. Based on your choice, we’ll give you some recommendations.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              autoFocus={true}
              name="title"
              onChange={[Function]}
              placeholder="Book Title"
              type="text"
              value=""
            />,
            <button
              className="search-btn"
            >
              Start your Search
            </button>,
          ],
          "className": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "name": "title",
              "onChange": [Function],
              "placeholder": "Book Title",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Start your Search",
              "className": "search-btn",
            },
            "ref": null,
            "rendered": "Start your Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="welcome text"
          >
            Welcome!
            <br />
            We’d like to help you find your next great read. Enter a book title you love, and click the search button to find that book. Based on your choice, we’ll give you some recommendations.
          </p>,
          <form
            className="search-form"
            onSubmit={[Function]}
          >
            <input
              autoFocus={true}
              name="title"
              onChange={[Function]}
              placeholder="Book Title"
              type="text"
              value=""
            />
            <button
              className="search-btn"
            >
              Start your Search
            </button>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome!",
              <br />,
              "We’d like to help you find your next great read. Enter a book title you love, and click the search button to find that book. Based on your choice, we’ll give you some recommendations.",
            ],
            "className": "welcome text",
          },
          "ref": null,
          "rendered": Array [
            "Welcome!",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "We’d like to help you find your next great read. Enter a book title you love, and click the search button to find that book. Based on your choice, we’ll give you some recommendations.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                autoFocus={true}
                name="title"
                onChange={[Function]}
                placeholder="Book Title"
                type="text"
                value=""
              />,
              <button
                className="search-btn"
              >
                Start your Search
              </button>,
            ],
            "className": "search-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "name": "title",
                "onChange": [Function],
                "placeholder": "Book Title",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Start your Search",
                "className": "search-btn",
              },
              "ref": null,
              "rendered": "Start your Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
